File: app/src/main/java/com/localphoto360/app/capture/Emulator.java
package com.localphoto360.app.capture;

import android.os.Build;

public final class Emulator {
    private Emulator() {
    }

    public static boolean isProbablyEmulator() {
        String fingerprint = Build.FINGERPRINT.toLowerCase();
        String model = Build.MODEL.toLowerCase();
        String product = Build.PRODUCT.toLowerCase();
        String hardware = Build.HARDWARE.toLowerCase();
        return fingerprint.contains("generic")
                || fingerprint.contains("emulator")
                || model.contains("emulator")
                || model.contains("android sdk")
                || product.contains("sdk")
                || hardware.contains("ranchu")
                || hardware.contains("goldfish");
    }
}

File: app/src/main/java/com/localphoto360/app/capture/Emulator.kt
package com.localphoto360.app.capture

import android.os.Build

fun isProbablyEmulator(): Boolean {
    val fingerprint = Build.FINGERPRINT.lowercase()
    val model = Build.MODEL.lowercase()
    val product = Build.PRODUCT.lowercase()
    val hardware = Build.HARDWARE.lowercase()
    return fingerprint.contains("generic") ||
        fingerprint.contains("emulator") ||
        model.contains("emulator") ||
        model.contains("android sdk") ||
        product.contains("sdk") ||
        hardware.contains("ranchu") ||
        hardware.contains("goldfish")
}
